backend/danswer/datastores/qdrant/store.py

```diff
@@ -81,32 +84,96 @@ def _build_qdrant_filters(
     return filter_conditions
 
 
+def _get_points_from_document_ids(
+    document_ids: list[str],
+    collection: str,
+    client: QdrantClient,
+) -> list[int | str]:
+    offset: int | str | None = 0
+    chunk_ids = []
+    while offset is not None:
+        matches, offset = client.scroll(
+            collection_name=collection,
+            scroll_filter=Filter(
+                must=[FieldCondition(key=DOCUMENT_ID, match=MatchAny(any=document_ids))]
+            ),
+            limit=_BATCH_SIZE,
+            with_payload=False,
+            with_vectors=False,
+            offset=offset,
+        )
+        for match in matches:
+            chunk_ids.append(match.id)
+
+    return chunk_ids
+
+
 class QdrantIndex(VectorIndex):
-    def __init__(self, collection: str = QDRANT_DEFAULT_COLLECTION) -> None:
-        self.collection = collection
+    def __init__(self, index_name: str = DOCUMENT_INDEX_NAME) -> None:
+        # In Qdrant, the vector index is referred to as a collection
+        self.collection = index_name
         self.client = get_qdrant_client()
 
+    def ensure_indices_exist(self) -> None:
+        if self.collection not in {
+            collection.name
+            for collection in list_qdrant_collections(self.client).collections
+        }:
+            logger.info(f"Creating Qdrant collection with name: {self.collection}")
+            create_qdrant_collection(
+                collection_name=self.collection, q_client=self.client
+            )
+
     def index(
         self,
-        chunks: list[EmbeddedIndexChunk],
+        chunks: list[IndexChunk],
         index_attempt_metadata: IndexAttemptMetadata,
-    ) -> list[ChunkInsertionRecord]:
+    ) -> set[DocumentInsertionRecord]:
         return index_qdrant_chunks(
             chunks=chunks,
             index_attempt_metadata=index_attempt_metadata,
             collection=self.collection,
             client=self.client,
         )
 
+    def update(self, update_requests: list[UpdateRequest]) -> None:
+        logger.info(
+            f"Updating {len(update_requests)} documents' allowed_users in Qdrant"
+        )
+        for update_request in update_requests:
+            for doc_id_batch in batch_generator(
+                items=update_request.document_ids,
+                batch_size=_BATCH_SIZE,
+            ):
+                chunk_ids = _get_points_from_document_ids(
+                    doc_id_batch, self.collection, self.client
+                )
+                self.client.set_payload(
```

Comment:
Note: we will be potentially updating quite a lot more than `_BATCH_SIZE` points here. Is that intended?

Reply:
_BATCH_SIZE is quite conservative, and this is a tiny request in terms of packet size so I figure it shouldn't be an issue. Plus this won't really be used anymore anyway...